### ISSUE TITLE
Allow the C compiler to be configurable on Windows

### ISSUE THREAD

**Issue by BrianOtto** [ziglang/zig#1314]
I am trying to use Zig as a C compiler, so that I can slowly replace a C library with Zig code.

I am running this on a Windows 10 machine, and it looks like Zig is hard-coded to run "cl.exe" when it detects the environment is `builtin.Environ.msvc`. This would normally be okay, but Zig tries to pass in parameters that this compiler doesn't accept (e.g. -fPIC, -fstack-protector-strong, etc).

See lines 635 - 637 in `getCCExe()` @ [std/build.zig](https://github.com/ziglang/zig/blob/02713e8d8aa9641616bd85e77dda784009c96113/std/build.zig)

Also see the parameters that get appended in `makeC()` when `self.kind = Kind.Lib`.

I think the proper solution would be to check for a "CC" environment variable first (see line 638), and if it doesn't exist then default to "cl.exe" on Windows (and when this is the case, then provide the proper parameters to it). As a quick fix, I would be happy if it just checked for "CC" first, as this would allow me to use gcc in TDM-GCC or MSYS on Windows, and then the parameters you are currently passing would work okay.

Here is the build file, in case it helps ...

```
const Builder = @import("std").build.Builder;

pub fn build(b: &Builder) void {
    b.addCIncludePath("src");
    
    const dyn = b.addCSharedLibrary("test", b.version(1, 0, 0));
    
    // multiple calls to dyn.addSourceFile()

    dyn.setOutputPath("bin");
    
    b.default_step.dependOn(&dyn.step);
}
```

**Comment by andrewrk** [comment]
I think this is #538 - `zig build` does not yet know how to be a C compiler driver on Windows.

**Comment by andrewrk** [comment]
This was fixed by #490
